### ISSUE TITLE
Withdrawals before Shanghai error message when syncing OP Sepolia

### ISSUE THREAD

**Issue by sealer3** [sunnyside-io/op-erigon#113]
I haven't been able to finish syncing OP Sepolia network due to this error. Does op-erigon not support OP Sepolia yet? It continues to sync until what corresponds to block [4089330](https://sepolia-optimistic.etherscan.io/block/4089330).

#### System information

Erigon version: erigon version 0.02.0-unstable

OS & Version: Ubuntu 22.04

Commit hash: 8cb1eb88f380103c910bc732448becb7c6e9b731

Erigon Command (with flags/config):
```
./build/bin/erigon --datadir=sepolia-data --genesis.path=op-sepolia-genesis.json --torrent.port=42068 --private.api.addr=127.0.0.1:9091 --http --ws --http.api=eth,debug,net,trace,web3,erigon --authrpc.jwtsecret=./jwt-secret --http.vhosts="*" --rollup.sequencerhttp=https://sepolia-sequencer.optimism.io/ --rollup.historicalrpc=https://sepolia.optimism.io/ --rollup.disabletxpoolgossip=true --nodiscover --maxpeers=0
```

Consensus Layer: op-node

Consensus Layer Command (with flags/config):
```
./op-node/bin/op-node \
        --l1=http://localhost:8545 \
        --l1.rpckind=debug_geth \
        --l2=http://localhost:8552 \
        --l2.jwt-secret=../op-erigon/jwt-secret \
        --network=op-sepolia \
        --rpc.addr=127.0.0.1 \
        --rpc.port=9545
```

Chain/Network: Optimistic Sepolia

#### Expected behaviour
Node should fully sync without errors.

#### Actual behaviour
In Erigon:
```
INFO[12-24|16:10:04.609] Received ForkchoiceUpdated [build]       version=2 head=0x2ea5dc85c900443e6ee9615876a413ef3f7f800792886e646f03cae258e30dc4 safe=0x2ea5dc85c900443e6ee9615876a413ef3f7f800792886e646f03cae258e30dc4 finalized=0x414150bd177495c6acea05ffb3a28655a28130e265201bf99a5f6dd85d66cd44 timestamp=0x6553a790 prevRandao=0x56c69fea1a23e350c8014171a2819ff4cc84a27272bae1a866d58b4ced467024 suggestedFeeRecipient=0x4200000000000000000000000000000000000011 withdrawals=[]
WARN[12-24|16:10:04.609] Served                                   conn=127.0.0.1:59148 method=engine_forkchoiceUpdatedV2 reqid=20459807 t=153.141us err="withdrawals before shanghai"
```

In op-node:
```
WARN [12-24|16:16:11.179] Failed to share forkchoice-updated signal state="&{HeadBlockHash:0x2ea5dc85c900443e6ee9615876a413ef3f7f800792886e646f03cae258e30dc4 SafeBlockHash:0x2ea5dc85c900443e6ee9615876a413ef3f7f800792886e646f03cae258e30dc4 FinalizedBlockHash:0x414150bd177495c6acea05ffb3a28655a28130e265201bf99a5f6dd85d66cd44}" err="withdrawals before shanghai"
WARN [12-24|16:16:11.179] Derivation process temporary error       attempts=3682 err="engine stage failed: temp: temporarily cannot insert new safe block: failed to create new block via forkchoice: unrecognized rpc error: withdrawals before shanghai"
```

#### Steps to reproduce the behaviour
Attempt sync with above commands

#### Backtrace
none

**Comment by pcw109550** [comment]
@sealer3 I assume that your`op-sepolia-genesis.json` does not have shanghai hard fork time/canyon hardfork time + config inited. If you did not specified necessary values for hard fork, you get a sync error.

Your genesis json's config key(only for op-sepolia) must be in the below form:
```
  "config": {
    "chainId": 11155420,
    "homesteadBlock": 0,
    ...
    "bedrockBlock": 0,
    "regolithTime": 0,
    "shanghaiTime": 1699981200,
    "canyonTime": 1699981200,
    "terminalTotalDifficulty": 0,
    "terminalTotalDifficultyPassed": true,
    "optimism": {
      "eip1559Elasticity": 6,
      "eip1559Denominator": 50,
      "eip1559DenominatorCanyon": 250
    }
  },
...
```

Note that `shanghaiTime`, `canyonTime` and `eip1559DenominatorCanyon` are added. Please update your json and resync. 

op-erigon technically supports syncing op-sepolia, but it is not officially supported: you may run into issues like this. We are planning to support it without custom config tweaks.

BTW, happy Christmas and happy hacking!

**Comment by sealer3** [comment]
Thank you so much for such a fast response! You're right - the official genesis json I got from following the OP Sepolia node operator tutorial must now be outdated since it did not include these fork times: https://docs.optimism.io/builders/node-operators/tutorials/testnet (which says to download https://networks.optimism.io/op-sepolia/genesis.json)

I've updated my genesis json with these new entries and will resync from the beginning. I'll come back and close this issue once it's synced :)

Merry Christmas to you too!
